akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -458,16 +458,19 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
 
   private lazy val shutdownHook = new Thread {
     override def run(): Unit = {
-      if (hasBeenShutdown.compareAndSet(false, true)) {
+      if (!hasBeenShutdown.get) {
```

Comment:
So this causes coordinated shutdown to happen also when the transport was already explicitly shut down? Sounds reasonable.